src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -115,12 +126,42 @@ protected Endpoint getEndpoint() {
         return Endpoint.INTERNALUSERS;
     }
 
+    @Override
+    protected void handleGet(final RestChannel channel, RestRequest request, Client client, final JsonNode content) throws IOException {
+        final String resourcename = request.param("name");
+
+        SecurityDynamicConfiguration<?> configuration = load(getConfigName(), true);
+        filter(configuration);
+
+        String includeIfType = request.param("filterBy", "any");
+
+        if (includeIfType != "any") {
+            try {
+                userService.includeAccountsIfType(configuration, AccountType.valueOf(includeIfType.toUpperCase()));
+            } catch (UserServiceException ex) {
+                badRequestResponse(channel, ex.getMessage());
+            }
+        }
+
+        // no specific resource requested, return complete config
+        if (Strings.isNullOrEmpty(resourcename)) {
+            successResponse(channel, configuration);
+            return;
+        }
+        if (!configuration.exists(resourcename)) {
+            notFound(channel, "Resource '" + resourcename + "' not found.");
+            return;
+        }
+        configuration.removeOthers(resourcename);
```

Comment:
Its strange that there are two different times when filtering is applied on this request's output, can this be better reconciled?

Reply:
This should become 'outdated' with the abstractApiAction changes from the refactor, but i believe it makes sense.  If someone Request JhonDoe's account (Which is an internal account) and ask to filter by service accounts,  it makes sense to me that it the account would not be displayed.